### PR TITLE
[WIP] Add Auth Code Request step to MSAL Electron PoC

### PR DESCRIPTION
**Pull request by technical-boy**

### CODE REVIEW

File: lib/msal-electron-proof-of-concept/src/AppConfig/ServerRequest/ServerRequestParameters.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { Authority } from "../Authority/Authority";
+import { timingSafeEqual } from "crypto";
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+export class ServerRequestParameters {
```
Comment (DarylThayil) [inline lib/msal-electron-proof-of-concept/src/AppConfig/ServerRequest/ServerRequestParameters.ts]: does this actually create a server request url?

Comment (technical-boy) [reply lib/msal-electron-proof-of-concept/src/AppConfig/ServerRequest/ServerRequestParameters.ts]: It was going to but not anymore. I’m changing it to an abstract class. Still a WIP right now. I’m going to use this as the base class for both Authorization and Token request parameters (both request URLs are different, but have stuff in common)

File: lib/msal-electron-proof-of-concept/src/AppConfig/AuthOptions.ts [hunk 2]
```diff
@@ -6,9 +6,11 @@
  * - clientId       - Client ID of your app registered in the application registration portal : https://portal.azure.com/#blade/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/RegisteredAppsPreview in Microsoft Identity Platform
  * - authority      - You can configure a specific authority, defaults to " " or "https://login.microsoftonline.com/common"
  * - redirectUri    - The redirect URI of the application, this should be the same as the value in the application registration portal. Defaults to `msal://auth`.
+ * - tenantId       - The Tenant ID of the application registered in the application registration portal.
  */
 export type AuthOptions = {
     clientId: string;
     authority?: string;
     redirectUri?: string;
+    tenantId: string;
```
Comment (sameerag) [inline lib/msal-electron-proof-of-concept/src/AppConfig/AuthOptions.ts]: Why do you need the tenantId and where would you need it?

File: lib/msal-electron-proof-of-concept/src/AppConfig/ClientApplication/ClientApplicationBase.ts [hunk 3]
```diff
@@ -1,8 +1,8 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-import { ApplicationConfiguration } from './ApplicationConfiguration';
-import { AuthOptions } from './AuthOptions';
+import { ApplicationConfiguration } from '../ApplicationConfiguration';
+import { AuthOptions } from '../AuthOptions';
 
 /**
  * ClientApplicationBase abstract class
```
Comment (sameerag) [inline lib/msal-electron-proof-of-concept/src/AppConfig/ClientApplication/ClientApplicationBase.ts]: I don't you need "Base" in the name as your inherited class is 'PublicClientApplication'. You can rename this 'ClientApplication' :)

File: lib/msal-electron-proof-of-concept/src/AppConfig/ClientApplication/PublicClientApplication.ts [hunk 4]
```diff
@@ -1,11 +1,13 @@
...
+import { ClientConfigurationError } from '../Error/ClientConfigurationError';
+import { ServerRequestParameters } from '../ServerRequest/ServerRequestParameters';
 import { ClientApplicationBase } from './ClientApplicationBase';
-import { ClientConfigurationError } from './Error/ClientConfigurationError';
 
 import { strict as assert } from 'assert';
+import { AadAuthority } from '../Authority/AadAuthority';
 
 /**
  * PublicClientApplication class
```
Comment (sameerag) [inline lib/msal-electron-proof-of-concept/src/AppConfig/ClientApplication/PublicClientApplication.ts]: I like the inheritance and simplification of each class. Good job. 👍

File: lib/msal-electron-proof-of-concept/src/AppConfig/ServerRequest/ServerRequestParameters.ts (hunk 1 above)
Comment (sameerag) [reply lib/msal-electron-proof-of-concept/src/AppConfig/ServerRequest/ServerRequestParameters.ts]: So the inherited class is yet to come?

**Pull request closed without merging**